Library/Homebrew/extend/os/linux/system_config.rb

```diff
@@ -0,0 +1,39 @@
+require "formula"
+
+class SystemConfig
+  class << self
+    def host_os_version
+      if which("lsb_release")
+        description = `lsb_release -d`.chomp.sub("Description:\t", "")
+        codename = `lsb_release -c`.chomp.sub("Codename:\t", "")
+        "#{description} (#{codename})"
+      elsif (redhat_release = Pathname.new("/etc/redhat-release")).readable?
+        redhat_release.read.chomp
+      else
+        "N/A"
+      end
+    end
+
+    def host_gcc_version
+      gcc = Pathname.new "/usr/bin/gcc"
+      `#{gcc} --version 2>/dev/null`[/ (\d+\.\d+\.\d+)/, 1] if gcc.exist?
+    end
+
+    def formula_version(formula)
+      f = Formula[formula]
+      f.installed? ? f.version : "N/A"
+    rescue FormulaUnavailableError
+      return "N/A"
+    end
+
+    def dump_verbose_config(f = $stdout)
+      dump_generic_verbose_config(f)
+      f.puts "Kernel: #{`uname -mors`.chomp}"
+      f.puts "OS: #{host_os_version}"
+      f.puts "OS gcc: #{host_gcc_version}"
+      f.puts "Linuxbrew glibc: #{formula_version "glibc"}"
```

Comment:
I prefer to use the name of the formula `glibc` as opposed to the upstream name.